server/helpers/format-user-data.js

```diff
@@ -27,5 +42,11 @@ module.exports = (goalsArr) => {
       }
     });
   });
+  return formattedGoals;
+};
+
+module.exports = (goalsArray) => {
+  let goals = removeDuplicates(formatGoalsArray(goalsArray));
+  goals = addRatings(goals, goalsArray);
   return JSON.stringify(goals);
 };
```

Comment:
`const formattedGoals` then `const goalsWithRatings` ? Something like that... avoids mutation and makes it clearer what's going on...